ext/opentelemetry-ext-opentracing-shim/src/opentelemetry/ext/opentracingshim/__init__.py

```diff
@@ -0,0 +1,257 @@
+        return ScopeWrapper(self, wrapped_span)
+        # TODO: Return a saved instance of SpanWrapper instead of constructing
+        # a new object (and the same for ScopeWrapper?).
+        # https://github.com/open-telemetry/opentelemetry-python/issues/161#issuecomment-534136274
+
+
+class TracerWrapper(opentracing.Tracer):
+    def __init__(self, tracer, scope_manager=None):
+        # If a scope manager isn't provided by the user, create a
+        # `ScopeManagerWrapper` instance and use it to initialize the
+        # `TracerWrapper`.
+        if scope_manager is None:
+            scope_manager = ScopeManagerWrapper(self)
+        super().__init__(scope_manager=scope_manager)
+        self._otel_tracer = tracer
+
+    def unwrap(self):
+        """Returns the wrapped OpenTelemetry `Tracer` object."""
+
+        return self._otel_tracer
+
+    @contextmanager
+    def start_active_span(
+        self,
+        operation_name,
+        child_of=None,
+        references=None,
+        tags=None,
+        start_time=None,
+        ignore_active_span=False,
+        finish_on_close=True,
+    ):
+        span = self.start_span(
+            operation_name=operation_name,
+            child_of=child_of,
+            references=references,
+            tags=tags,
+            start_time=start_time,
+            ignore_active_span=ignore_active_span,
+        )
+        with self._scope_manager.activate(span, finish_on_close) as scope:
+            yield scope
+
+    def start_span(
+        self,
+        operation_name=None,
+        child_of=None,
+        references=None,
+        tags=None,
+        start_time=None,
+        ignore_active_span=False,
+    ):
+        if child_of is None:
+            parent = None
+        else:
+            # TODO: Should we use the OpenTracing base classes for the type
+            # check?
+            if isinstance(child_of, (SpanWrapper, SpanContextWrapper)):
+                # The parent specified in `child_of` is valid and is either a
+                # `SpanWrapper` or a `SpanContextWrapper`. Unwrap the `Span` or
+                # `SpanContext` to extract the OpenTracing object and use this
+                # object as the parent of the created span.
+                parent = child_of.unwrap()
+            else:
+                logger.warning(
+                    "Unknown class %s passed in child_of argument to start_span() method.",
+                    type(child_of),
+                )
+                parent = None
+                # TODO: Refuse to create a span and return `None` instead of
+                # proceeding with a `None` parent? This would cause the created
+                # span to become a child of the active span, if any, or create
+                # a new trace and make the span the root span of that trace.
+
+        # Use active span as parent when no explicit parent is specified.
+        if (
+            self.active_span is not None
```

Comment:
I feel like we would need to create/discard a wrapper here, as we would be using the OT api, not the OTel one - that being said, I don't see the `active_span` definition anywhere.

Reply:
@carlosalberto `active_span` is a property on `opentracing.Tracer`:

https://github.com/opentracing/opentracing-python/blob/7d2e62b5999d3cdec9a9c2720a34a1d11f425461/opentracing/tracer.py#L50-L60

This property returns the span returned in `self._scope_manager.active`, which is fine for us because in our implementation of the `active` property we return a `ScopeWrapper` with a `SpanWrapper` inside of it.

Since `active_span` has a tiny implementation inside the OT API which fits our needs I didn't see a need to override it.